syntax = "proto3";

// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. chat_service.proto

package chat;

service ChatService {

  rpc JoinChat (JoinRequest) returns (stream ChatMessage) {}

  rpc SendMessage (SendRequest) returns (Empty) {}
}

message JoinRequest {
  string username = 1;
}

message SendRequest {
  string sender_username = 1;
  string recipient_username = 2;
  string content = 3;
}

message Empty {}

message ChatMessage {
  string sender_username = 1;
  string content = 2;
}

// Auth Service

service AuthService {
    rpc Login (LoginRequest) returns (LoginResponse) {}
    rpc Register (RegisterRequest) returns (RegisterResponse) {}
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    bool success = 1;
    string message = 2;
}

message RegisterRequest {
    string username = 1;
    string password = 2;
}

message RegisterResponse {
    bool success = 1;
    string message = 2;
}